azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java

```diff
@@ -439,6 +448,32 @@ public void run() {
                             AmqpConstants.OFFSET_ANNOTATION_NAME,
                             offsetInclusiveFlag ? "=" : StringUtil.EMPTY,
                             lastReceivedOffset));
+        } else {
+            final String lastReceivedSequenceNumber;
+            final boolean sequenceNumberInclusiveFlag;
+            if (lastReceivedMessage != null) {
+                sequenceNumberInclusiveFlag = false;
+                lastReceivedSequenceNumber = lastReceivedMessage.getMessageAnnotations().getValue().get(AmqpConstants.SEQUENCE_NUMBER).toString();
```

Comment:
lookup on Sequence_number is expensive on eventhubs service. we are optimized for Offset. so, whenever we have the detail on LastReceivedMessage - use offset based receiveLink.

So, this if...else block will be more readable if it is organized as:
```
if (lastReceivedMessage == null) {
 if (this.startingDateTime is specified) {  //receiveByStargingDateTime  }
 else if (this.startingseqNo is specified) {   ...... }
 else // if (this.offset is specified) {...}
}
else {
 // receiveByLastReceivedMessage offset
}

```